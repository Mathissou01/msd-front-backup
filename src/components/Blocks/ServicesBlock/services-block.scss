@use "src/styles/02-tools/tools.include-media" as im;
@use "src/styles/01-settings/settings.colors" as colors;
@use "src/styles/02-tools/tools.z-index" as index;

.c-ServicesBlock {
  position: relative;
  display: flex;
  justify-content: stretch;
  min-height: 268px;
  padding: 40px 16px 60px;
  overflow: hidden;

  @include im.media(">=mobile") {
    padding: 60px;
  }

  @include im.media(">=laptop") {
    justify-content: center;
    padding: 80px 60px;
  }

  @include im.media(">=desktop") {
    padding: 80px 124px;
  }

  &__Content {
    position: relative;
    z-index: index.z("site-content");
    display: flex;
    flex-direction: column;
    gap: 24px;
    align-items: flex-start;
    width: 100%;

    @include im.media(">=tablet") {
      max-width: none;
    }
  }

  &__Services {
    display: flex;
    flex-flow: column wrap;
    gap: 14px;
    width: 100%;

    @include im.media(">=tablet") {
      flex-direction: row;
      align-self: center;
      max-width: 985px;

      > * {
        flex-basis: calc((100% - 28px) / 3);
      }
    }
  }

  &__Card {
    display: flex;
    overflow: hidden;
    border-radius: 8px;

    @include im.media(">=tablet", "<laptop") {
      flex-basis: calc((100% - 14px) / 2);
    }

    @include im.media(">=laptop") {
      flex-basis: calc((100% - 28px) / 3);
    }

    &:hover {
      background-color: colors.$white;
    }

    &_few {
      flex-basis: calc((100% - 14px) / 2);
    }
  }

  &__SvgContainer {
    position: absolute;
    bottom: 0;
    left: 0;
    z-index: index.z("background");
    width: 100%;
    background: colors.$background-dark;
    pointer-events: none;

    @include im.media(">=tablet") {
      bottom: -1px;
      background: colors.$background-dark;
      border-radius: 0 120px 0 0;
    }
  }

  &__Svg {
    &_mobile {
      display: initial;

      @include im.media(">=tablet") {
        display: none;
      }
    }

    &_desktop {
      display: none;

      @include im.media(">=tablet") {
        display: initial;
      }
    }
  }

  svg {
    height: 142px;
    margin-top: -21px;

    @include im.media(">=laptop") {
      height: 165px;
    }

    #primary-color {
      fill: colors.$brand-primary;
    }

    #secondary-color {
      fill: colors.$brand-secondary;
    }
  }
}
