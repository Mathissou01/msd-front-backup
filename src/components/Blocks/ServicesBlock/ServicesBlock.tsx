import classNames from "classnames";
import { IServiceLink } from "../../../lib/service-links";
import CommonBlockHeading from "../../Common/CommonBlockHeading/CommonBlockHeading";
import ServiceCard from "./ServiceCard/ServiceCard";
import ServicesMobile from "public/images/mobile_services-section_bottom-left-angle.svg";
import ServicesDesktop from "public/images/desktop_services-section_bottom-left-angle.svg";
import "./services-block.scss";

interface IServicesBlockProps {
  remappedData: {
    titleContent: string | null;
    serviceLinks: Array<IServiceLink> | null;
  };
}

export default function ServicesBlock({ remappedData }: IServicesBlockProps) {
  /* Static Data */
  const defaultPicto = "/images/pictos/default.svg";

  /* Local Data */
  const countVisibleServices =
    remappedData.serviceLinks?.filter((service) => service.isDisplayed)
      .length ?? 0;

  return (
    <section className="c-ServicesBlock" data-testid="services-block">
      <div className="c-ServicesBlock__Content">
        <CommonBlockHeading titleContent={remappedData.titleContent ?? ""} />
        <div className="c-ServicesBlock__Services">
          {remappedData.serviceLinks?.map(
            (link, index) =>
              link.isDisplayed && (
                <div
                  key={`${link.name}_${index}`}
                  className={classNames("c-ServicesBlock__Card", {
                    "c-ServicesBlock__Card_few": countVisibleServices < 3,
                  })}
                >
                  <ServiceCard
                    href={
                      link.externalLink
                        ? link.externalLink
                        : link.generatedSlug ?? "/"
                    }
                    name={link.name ?? ""}
                    pictoUrl={link.picto?.data?.attributes?.url ?? defaultPicto}
                  />
                </div>
              ),
          )}
        </div>
      </div>
      <div className="c-ServicesBlock__SvgContainer">
        <ServicesDesktop className="c-ServicesBlock__Svg_desktop" />
        <ServicesMobile className="c-ServicesBlock__Svg_mobile" />
      </div>
    </section>
  );
}
