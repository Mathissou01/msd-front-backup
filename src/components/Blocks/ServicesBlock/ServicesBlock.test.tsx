import { render, screen } from "@testing-library/react";
import {
  defaultMockData,
  emptyMockData,
  lessThanThreeMockData,
} from "../../../../__mocks__/ServicesBlockMockData";
import ServicesBlock from "./ServicesBlock";

describe("ServicesBlock", () => {
  it("renders nothing when data don't contain a required data ", () => {
    const { container } = render(
      <ServicesBlock remappedData={emptyMockData} />,
    );
    const component = screen.queryByTestId("services-block");
    expect(component).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it("renders less than three ServiceCard", () => {
    const { container } = render(
      <ServicesBlock remappedData={lessThanThreeMockData} />,
    );
    const OneService = screen.getByText("Guide du tri");
    expect(OneService).toBeInTheDocument();

    const cards = screen.getAllByTestId("service-card");
    expect(cards.length).toBe(2);
    expect(container).toMatchSnapshot();
  });

  it("renders three or more of ServiceCard", () => {
    const { container } = render(
      <ServicesBlock remappedData={defaultMockData} />,
    );
    const OneService = screen.getByText("Réduire mes déchets");
    expect(OneService).toBeInTheDocument();

    const cards = screen.getAllByTestId("service-card");
    expect(cards.length).toBe(4);
    expect(container).toMatchSnapshot();
  });
});
