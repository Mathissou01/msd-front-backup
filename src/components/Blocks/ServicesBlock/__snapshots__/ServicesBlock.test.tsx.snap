// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ServicesBlock renders less than three ServiceCard 1`] = `
<div>
  <section
    class="c-ServicesBlock"
    data-testid="services-block"
  >
    <div
      class="c-ServicesBlock__Content"
    >
      <div
        class="c-CommonBlockHeading"
      >
        <h2>
          Les services de ma ville
        </h2>
        <div
          class="c-CommonBlockHeading__Line"
        />
      </div>
      <div
        class="c-ServicesBlock__Services"
      >
        <div
          class="c-ServicesBlock__Card c-ServicesBlock__Card_few"
        >
          <a
            class="c-ServiceCard"
            data-testid="service-card"
            href="/"
          >
            <div
              class="c-ServiceCard__Svg"
            >
              <div
                class="c-ServiceCard__Picto"
              >
                <img
                  alt=""
                  data-nimg="1"
                  decoding="async"
                  height="25"
                  loading="lazy"
                  src="/images/pictos/default.svg"
                  style="color: transparent;"
                  width="25"
                />
              </div>
            </div>
            <div
              class="c-ServiceCard__Name"
            >
              Guide du tri
            </div>
          </a>
        </div>
        <div
          class="c-ServicesBlock__Card c-ServicesBlock__Card_few"
        >
          <a
            class="c-ServiceCard"
            data-testid="service-card"
            href="valorisation-de-mes-dechets"
          >
            <div
              class="c-ServiceCard__Svg"
            >
              <div
                class="c-ServiceCard__Picto"
              >
                <img
                  alt=""
                  data-nimg="1"
                  decoding="async"
                  height="25"
                  loading="lazy"
                  src="/images/pictos/default.svg"
                  style="color: transparent;"
                  width="25"
                />
              </div>
            </div>
            <div
              class="c-ServiceCard__Name"
            >
              Valorisation de mes déchets
            </div>
          </a>
        </div>
      </div>
    </div>
    <div
      class="c-ServicesBlock__SvgContainer"
    >
      <div
        class="c-ServicesBlock__Svg_desktop"
      />
      <div
        class="c-ServicesBlock__Svg_mobile"
      />
    </div>
  </section>
</div>
`;

exports[`ServicesBlock renders nothing when data don't contain a required data  1`] = `
<div>
  <section
    class="c-ServicesBlock"
    data-testid="services-block"
  >
    <div
      class="c-ServicesBlock__Content"
    >
      <div
        class="c-CommonBlockHeading"
      >
        <h2>
          Les services de ma ville
        </h2>
        <div
          class="c-CommonBlockHeading__Line"
        />
      </div>
      <div
        class="c-ServicesBlock__Services"
      />
    </div>
    <div
      class="c-ServicesBlock__SvgContainer"
    >
      <div
        class="c-ServicesBlock__Svg_desktop"
      />
      <div
        class="c-ServicesBlock__Svg_mobile"
      />
    </div>
  </section>
</div>
`;

exports[`ServicesBlock renders three or more of ServiceCard 1`] = `
<div>
  <section
    class="c-ServicesBlock"
    data-testid="services-block"
  >
    <div
      class="c-ServicesBlock__Content"
    >
      <div
        class="c-CommonBlockHeading"
      >
        <h2>
          Les services de ma ville
        </h2>
        <div
          class="c-CommonBlockHeading__Line"
        />
      </div>
      <div
        class="c-ServicesBlock__Services"
      >
        <div
          class="c-ServicesBlock__Card"
        >
          <a
            class="c-ServiceCard"
            data-testid="service-card"
            href="/"
          >
            <div
              class="c-ServiceCard__Svg"
            >
              <div
                class="c-ServiceCard__Picto"
              >
                <img
                  alt=""
                  data-nimg="1"
                  decoding="async"
                  height="25"
                  loading="lazy"
                  src="/images/pictos/default.svg"
                  style="color: transparent;"
                  width="25"
                />
              </div>
            </div>
            <div
              class="c-ServiceCard__Name"
            >
              Guide du tri
            </div>
          </a>
        </div>
        <div
          class="c-ServicesBlock__Card"
        >
          <a
            class="c-ServiceCard"
            data-testid="service-card"
            href="valorisation-de-mes-dechets"
          >
            <div
              class="c-ServiceCard__Svg"
            >
              <div
                class="c-ServiceCard__Picto"
              >
                <img
                  alt=""
                  data-nimg="1"
                  decoding="async"
                  height="25"
                  loading="lazy"
                  src="/images/pictos/default.svg"
                  style="color: transparent;"
                  width="25"
                />
              </div>
            </div>
            <div
              class="c-ServiceCard__Name"
            >
              Valorisation de mes déchets
            </div>
          </a>
        </div>
        <div
          class="c-ServicesBlock__Card"
        >
          <a
            class="c-ServiceCard"
            data-testid="service-card"
            href="reduire-mes-dechets"
          >
            <div
              class="c-ServiceCard__Svg"
            >
              <div
                class="c-ServiceCard__Picto"
              >
                <img
                  alt=""
                  data-nimg="1"
                  decoding="async"
                  height="25"
                  loading="lazy"
                  src="/images/pictos/default.svg"
                  style="color: transparent;"
                  width="25"
                />
              </div>
            </div>
            <div
              class="c-ServiceCard__Name"
            >
              Réduire mes déchets
            </div>
          </a>
        </div>
        <div
          class="c-ServicesBlock__Card"
        >
          <a
            class="c-ServiceCard"
            data-testid="service-card"
            href="/"
          >
            <div
              class="c-ServiceCard__Svg"
            >
              <div
                class="c-ServiceCard__Picto"
              >
                <img
                  alt=""
                  data-nimg="1"
                  decoding="async"
                  height="25"
                  loading="lazy"
                  src="/images/pictos/default.svg"
                  style="color: transparent;"
                  width="25"
                />
              </div>
            </div>
            <div
              class="c-ServiceCard__Name"
            >
              dfhsrgsrh
            </div>
          </a>
        </div>
      </div>
    </div>
    <div
      class="c-ServicesBlock__SvgContainer"
    >
      <div
        class="c-ServicesBlock__Svg_desktop"
      />
      <div
        class="c-ServicesBlock__Svg_mobile"
      />
    </div>
  </section>
</div>
`;
